lib/puppetx/simp/compliance_mapper.rb

```diff
@@ -123,15 +123,9 @@ def enforcement(key, context=self, options={"mode" => "value"}, &block)
 
 # These cache functions are assumed to be created by the wrapper
 # object backend.
+# Caching disabled temporarily (SIMP-9623).
 def cached_lookup(key, default, &block)
-  if cache_has_key(key)
-    retval = cached_value(key)
-  else
-    retval = yield key, default
-    cache(key, retval)
-  end
-
-  retval
+  yield key, default
```

Comment:
I noticed that `cached_lookup()` is also called in the `load()`  method on lines 155 & 156, but those calls weren't replaced with `call_function('lookup'` like it was in line 51.

So: I'm wondering if `cached_lookup()`
should perform the `call_function('lookup'` here instead of simply yielding. 🤔

Reply:
@silug Let's try what @op-ct is suggesting and see if it fixes the issue. I'm not a fan of disabling caching if we can avoid it (for obvious reasons).

Reply:
To be clear: what I was suggesting it wouldn't keep caching on, because it would still perform a direct `lookup`.  I was just curious why the one call on 51 was replaced with an un-cached `lookup` and the others on 155–156 weren't.

And on second look, the answer to my question was clearly "no, because the lookups on line 51 and 155–6 are actually looking up different things ('compliance_markup::enforcement'` vs `compliance_markup::compliance_map` vs `compliance_map`). 

But I'm still curious if they should be replaced with the same `call_function('lookup',` pattern, because it looks like they won't receive anything with `cached_lookup` just yielding its own arguments.

Reply:
@op-ct Honestly my first thought was to just drop any caching on the result of `compliance_markup::enforcement` from Hiera, which is why I replaced the `cached_lookup` on line 51.  I later ripped out the rest of the caching when I realized it probably wasn't doing anything good.

Reply:
(Given the other changes, I'm fine with reverting the change on that one line if anyone cares.)

Reply:
@silug We should probably drop all caching across the board.